ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala

```diff
@@ -246,6 +249,102 @@ private[appendonlydao] final class TransactionsReader(
       .map(EventsTable.Entry.toGetTransactionResponse)
   }
 
+  override def getTransactionLogUpdates(
+      startExclusive: (Offset, Long),
+      endInclusive: (Offset, Long),
+  )(implicit
+      loggingContext: LoggingContext
+  ): Source[((Offset, Long), TransactionLogUpdate), NotUsed] = {
+    val endMarker = Source.single(
+      endInclusive -> TransactionLogUpdate.LedgerEndMarker(
+        eventOffset = endInclusive._1,
+        eventSequentialId = endInclusive._2,
+      )
+    )
+
+    val eventsSource = Source
+      .fromIterator(() =>
+        splitRange(
+          startExclusive._2,
+          endInclusive._2,
+          TransactionEventsFetchParallelism,
+        ).iterator
+      )
+      .mapAsync(TransactionEventsFetchParallelism) { range =>
+        dispatcher.executeSql(dbMetrics.getTransactionLogUpdates) { implicit conn =>
+          QueryNonPruned.executeSqlOrThrow(
+            query = TransactionLogUpdatesReader.readRawEvents(range),
+            minOffsetExclusive = startExclusive._1,
+            error = pruned =>
+              s"Active contracts request after ${startExclusive._1.toHexString} precedes pruned offset ${pruned.toHexString}",
+          )
+        }
+      }
+      .flatMapConcat(v => Source.fromIterator(() => v.iterator))
+      .mapAsync(TransactionEventsFetchParallelism) { raw =>
+        Timed.future(
+          metrics.daml.index.decodeTransactionLogUpdate,
+          Future(TransactionLogUpdatesReader.toTransactionEvent(raw)),
+        )
+      }
+
+    InstrumentedSource
+      .bufferedSource(
+        original = groupContiguous(eventsSource)(by = _.transactionId)
+          .map { v =>
+            val tx = toTransaction(v)
+            (tx.offset, tx.lastEventSequentialId) -> tx
+          }
+          .mapMaterializedValue(_ => NotUsed),
+        counter = metrics.daml.index.transactionLogUpdatesBufferSize,
+        size = outputStreamBufferSize,
+      )
+      .concat(endMarker)
+  }
+
+  private def splitRange(
+      startExclusive: Long,
+      endInclusive: Long,
+      numberOfChunks: Int,
+  ): Seq[EventsRange[Long]] =
```

Comment:
A few unit tests for this method would be great.

Reply:
Yeah, why not, good practice. Added tests